akka-projection-eventsourced/src/main/scala/akka/projection/eventsourced/javadsl/EventSourcedProvider.scala

```diff
@@ -44,7 +45,8 @@ object EventSourcedProvider {
       system: ActorSystem[_],
       eventsByTagQuery: EventsByTagQuery,
       tag: String)
-      extends SourceProvider[Offset, EventEnvelope[Event]] {
+      extends javadsl.SourceProvider[Offset, EventEnvelope[Event]]
+      with javadsl.CreationTimeSourceProvider[Offset, EventEnvelope[Event]] {
```

Comment:
In the InternalProjectionState where this will be used we don't want to look for two interfaces javadsl or scaladsl, so this doesn't work well. The `javadsl.CreationTimeSourceProvider` could extend `scaladsl.CreationTimeSourceProvider` or we just make one trait that is common for both javadsl and scaladsl since there is no difference in the signature

Reply:
I was wrestling with this as well.  Since the `SourceProvider` base type itself exists for Scala and Java it requires separate implementations for the `SourceProvider` behaviours that extend each, even if there's no type differences to warrant it.

I suppose we don't have to subclass the base `SourceProvider`, and can just add them to the root projections API.

Reply:
Oh, I think you're suggesting we move `CreationTimeSourceProvider` out of scala/java DSL entirely.

Reply:
@patriknw / @ignasi35 I'm going to drop the inheritance between the `SourceProvider` behaviours, unless there's another idea for how to proceed here.

Reply:
@seglo maybe it's not important to have the optional capability interface for this and we can just define it on the regular SourceProvider?

Reply:
It's not that important, it just would have been cleaner IMO.  I've added it to the regular `SourceProvider`.  I'm off this week, but @ignasi35 said he will iterate the PR further if necessary.